src/blocking/spi.rs

```diff
@@ -102,3 +102,52 @@ pub mod write_iter {
         }
     }
 }
+
+/// Operation for transactional SPI trait
+///
+/// This allows composition of SPI operations into a single bus transaction
+#[derive(Debug, PartialEq)]
+pub enum Operation<'a, W: 'static> {
+    /// Write data from the provided buffer, discarding read data
+    Write(&'a [W]),
+    /// Write data out while reading data into the provided buffer
+    Transfer(&'a mut [W]),
+}
+
+/// Transactional trait allows multiple actions to be executed
+/// as part of a single SPI transaction
+pub trait Transactional<W: 'static> {
+    /// Associated error type
+    type Error;
+
+    /// Execute the provided transactions
+    fn try_exec<'a>(&mut self, operations: &mut [Operation<'a, W>]) -> Result<(), Self::Error>;
+}
+
+/// Blocking transactional impl over spi::Write and spi::Transfer
+pub mod transactional {
+    use super::{Operation, Transfer, Write};
+
+    /// Default implementation of `blocking::spi::Transactional<W>` for implementers of
+    /// `spi::Write<W>` and `spi::Transfer<W>`
+    pub trait Default<W, E> {}
+
+    impl<W: 'static, E, S> super::Transactional<W> for S
+    where
+        S: self::Default<W, E> + Write<W, Error = E> + Transfer<W, Error = E>,
+        W: Copy + Clone,
+    {
+        type Error = E;
+
+        fn try_exec<'a>(&mut self, operations: &mut [super::Operation<'a, W>]) -> Result<(), E> {
```

Comment:
Should this function be named `try_commit`? :)

Reply:
i am open to renaming if there's a better option but don't know that `commit` is necessarily more clear than `exec`?

Reply:
Don't know if this is proper, but in database operations we often pair 'transaction' with 'commit'. Or this word is shorter than 'execute'. (I'm open to a more proper name)

Reply:
I think commit, exec, or submit are fine.  The kernel docs talk about "submitting" a spi message or transaction to the kernel.

Reply:
Maybe 'submit' is better (in my opinion). Database's 'commit' would somehow remind people that that's an operation which can be reverted atomatically. If all articles on SPI talks about 'submit', I personally think this could be yhe best word.

Reply:
`submit` feels like it implies there's a kernel to which the operation is passed, which is not always the case (and as it's a blocking API, you're not going to hear back until it's executed either way).